src/cc/bcc_elf.c

```diff
@@ -738,6 +738,24 @@ int bcc_elf_get_type(const char *path) {
     return hdr.e_type;
 }
 
+uint64_t bcc_elf_get_base_address(const char *path) {
+  Elf *e;
+  GElf_Phdr phdr;
+  GElf_Addr base_address;
+  int fd;
+
+  if (openelf(path, &e, &fd) < 0)
+    return -1;
+
+  gelf_getphdr(e, 0, &phdr);
+
+  base_address = phdr.p_vaddr-phdr.p_offset;
```

Comment:
The base_address is computed based on index 0 which is typically PHDR. Does this calculated base_address is always the same for the loaded segment?

Reply:
Sorry, what do you mean the loaded segment? Could you help to show an example? Thanks!

Reply:
This is what I mean:
```
-bash-4.4$ cat main.c
int main() { return 0; }
-bash-4.4$ gcc main.c
-bash-4.4$ readelf -l a.out

Elf file type is EXEC (Executable file)
Entry point 0x4003e0
There are 9 program headers, starting at offset 64

Program Headers:
  Type           Offset             VirtAddr           PhysAddr
                 FileSiz            MemSiz              Flags  Align
  PHDR           0x0000000000000040 0x0000000000400040 0x0000000000400040
                 0x00000000000001f8 0x00000000000001f8  R E    8
  INTERP         0x0000000000000238 0x0000000000400238 0x0000000000400238
                 0x000000000000001c 0x000000000000001c  R      1
      [Requesting program interpreter: /lib64/ld-linux-x86-64.so.2]
  LOAD           0x0000000000000000 0x0000000000400000 0x0000000000400000
                 0x000000000000069c 0x000000000000069c  R E    200000
  LOAD           0x0000000000000e10 0x0000000000600e10 0x0000000000600e10
                 0x0000000000000214 0x0000000000000218  RW     200000

```
We probably care about the first LOAD segment, right? In this case, the `virtual_address - offset` is the same. I am just wondering whether this is true in general.

Reply:
Actually I am not sure if this is the same in all cases. So how about update the code with below part?
```
for (i = 0; i < phdr_num; i++) {
    if (!gelf_getphdr(e, (int)i, &phdr))
      continue;
    if (phdr.p_type == PT_LOAD) {
      base_address = phdr.p_vaddr-phdr.p_offset;
      break;
    }
  }
```